src/libraries/System.Threading/tests/MutexTests.cs

```diff
@@ -349,7 +350,7 @@ public void OpenExisting_InvalidNames()
             AssertExtensions.Throws<ArgumentException>("name", null, () => Mutex.OpenExisting(string.Empty, options: default));
         }
 
-        [Fact]
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.HasAssemblyFiles))]
```

Comment:
I'm a bit worried about the number of tests that are getting disabled in this PR by: 
`public static bool HasAssemblyFiles => !string.IsNullOrEmpty(typeof(PlatformDetection).Assembly.Location);`

Do we have a clear picture of what platforms are under this check as I suppose it is not just Android.

Reply:
When assemblies are loaded from memory or embedded in a single‐file:
 - nativeaot / single‐file
 - browser / wasm
 - coreclr android
 - apple mobile with nativeaot

Reply:
Let's ask someone from wasm and nativeaot if they are okay with the disabling the tests on their platforms.

Reply:
I’ve verified that the tests disabled in this PR were already skipped on Native AOT and other mobile platforms. The failures unique to CoreCLR Android are:
```
Microsoft.VisualBasic.Tests.ErrObjectTests.Clear
MonoTests.System.Configuration.mapped_ExeConfiguration_null
MonoTests.System.Configuration.OpenExeConfiguration1_UserLevel_None
```